add information on the system type string (#8315)
NixOS/nix: doc/manual/src/language/builtin-constants.md

- [`builtins.currentSystem`]{#builtins-currentSystem} (string)

>   Return the system type of the Nix installation on which the expression is being evaluated.
>   This value is determined by the [`system` configuration setting](@docroot@/command-ref/conf-file.md#conf-system).

Reviewer: Problems:
 - `system` doesn't _have_ to be the system of the Nix installation.
 - values don't return anything.

```suggestion
  The system type of the Nix installation on which the expression is being evaluated, or the [`system` configuration setting](@docroot@/command-ref/conf-file.md#conf-system) if set to a different value.
```

Did I keep this in the order of relevance or did I make it contrived?
At least it's more truthful.

Author: ```suggestion
  The value of the [`system` configuration setting](@docroot@/command-ref/conf-file.md#conf-system).
```
